### PR TITLE
etcdhttp/metrics.go: exclude alarms from health check conditionally with `?exclude=NOSPACE`

### PR DESCRIPTION
**Pull request by chaochn47**
### Overview
Exclude alarms from health check with `?exclude=NOSPACE`

Similar to kubernetes excludes specific checks 
https://kubernetes.io/docs/reference/using-api/health-checks/#api-endpoints-for-health

```
curl -k 'https://localhost:6443/readyz?verbose&exclude=etcd'
```

The motivation is to exclude NOSPACE alarms in etcd supervisor or peer to peer health checks.

### Testing
Added unit test against serverV2 to verify the logic due to v3 `*etcdserver.EtcdServer` is hard to mock out. 

Unit test output
```
go test -v -run TestHealthHandler -mod=readonly
=== RUN   TestHealthHandler
{"level":"debug","msg":"serving /health true"}
{"level":"debug","msg":"/health OK","status-code":200}
{"level":"warn","msg":"serving /health false due to an alarm","alarm":"alarm:NOSPACE "}
{"level":"warn","msg":"/health error","output":"{\"health\":\"false\",\"reason\":\"ALARM NOSPACE\"}","status-code":503}
{"level":"debug","msg":"/health excluded alarm","alarm":"NOSPACE"}
{"level":"debug","msg":"serving /health true"}
{"level":"debug","msg":"/health OK","status-code":200}
{"level":"warn","msg":"fail exclude alarms from health check","exclude alarms":"map[NOSPACE:{}]"}
{"level":"debug","msg":"serving /health true"}
{"level":"debug","msg":"/health OK","status-code":200}
{"level":"debug","msg":"/health excluded alarm","alarm":"NOSPACE"}
{"level":"warn","msg":"serving /health false due to an alarm","alarm":"memberID:1 alarm:CORRUPT "}
{"level":"warn","msg":"/health error","output":"{\"health\":\"false\",\"reason\":\"ALARM CORRUPT\"}","status-code":503}
{"level":"debug","msg":"/health excluded alarm","alarm":"NOSPACE"}
{"level":"debug","msg":"/health excluded alarm","alarm":"CORRUPT"}
{"level":"debug","msg":"serving /health true"}
{"level":"debug","msg":"/health OK","status-code":200}
--- PASS: TestHealthHandler (0.01s)
PASS
ok      go.etcd.io/etcd/server/v3/etcdserver/api/etcdhttp       0.120s
```

### CODE REVIEW

**Comment by chaochn47** [comment]
@gyuho PTAL, thanks!!

File: server/etcdserver/api/etcdhttp/metrics.go [hunk 1]
```diff
@@ -110,26 +111,52 @@ type Health struct {
...
-			case etcdserverpb.AlarmType_NOSPACE:
-				h.Reason = "ALARM NOSPACE"
-			case etcdserverpb.AlarmType_CORRUPT:
-				h.Reason = "ALARM CORRUPT"
-			default:
-				h.Reason = "ALARM UNKNOWN"
+			alarmName := v.Alarm.String()
+			if _, found := excludedAlarms[alarmName]; found {
+				lg.Debug("/health excluded alarm", zap.String("alarm", alarmName))
+				delete(excludedAlarms, alarmName)
```
Comment (gyuho) [inline server/etcdserver/api/etcdhttp/metrics.go:141]: No need to delete? And we can just

```go
if _, found := excludedAlarms[alarmName]; found {
   ...
   continue;
}

h.Health = "false"
...
```

Comment (chaochn47) [reply server/etcdserver/api/etcdhttp/metrics.go:141]: Make sense. Done with next revision.

**Comment by chaochn47** [comment]
DP update:


> [CP / DP] New platform versions missing from cluster count graphs


CR send out to update 1.15 - 1.19 PV https://code.amazon.com/reviews/CR-49100790


________________________________

**Review by gyuho — APPROVED** [review]
lgtm

**Review by hexfusion — APPROVED** [review]

**Pull request merged**